packages/block-library/src/gallery/edit.js

```diff
@@ -339,7 +340,7 @@ function GalleryEdit( props ) {
 			accept="image/*"
 			allowedTypes={ ALLOWED_MEDIA_TYPES }
 			multiple
-			value={ images }
+			value={ hasImageIds ? images : {} }
```

Comment:
if we go with this approach I would maybe go with `hasImageIds ? images : undefined` as `MediaPlaceholder` sets the default to `{}` so avoids an issue if this default changes